### PR TITLE
[FLINK-10261][table] fix insert into with order by

### PR DESCRIPTION
**Pull request by xueyumusic**
## What is the purpose of the change
This PR tried to fix insert into with order by. It was caused by calcite validate rewrite and change SqlNode in some cases such as ORDER_BY sql kind. In these special cases validate return new SqlNode with different kind, however, flink does not use this new SqlNode.
In addition, `DataStreamSortRule` seems require the first sort attributes is time attributes and order is ascending

## Brief change log
TableEnvironment validate

## Verifying this change

test case

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/sql/validation/InsertIntoValidationTest.scala [hunk 1]
```diff
@@ -47,6 +48,27 @@ class InsertIntoValidationTest {
     tEnv.sqlUpdate(sql)
   }
 
+  @Test(expected = classOf[TableException])
+  def testNonTimeOrderByInsert(): Unit = {
```
[the remark this answers is not in the document]
Comment (xueyumusic) [reply flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/sql/validation/InsertIntoValidationTest.scala]: yes, SortValidationTest has similar time field order tests

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/sql/SqlITCase.scala [hunk 2]
```diff
@@ -742,6 +742,36 @@ class SqlITCase extends StreamingWithStateTestBase {
...
+    val sql = "INSERT INTO targetTable SELECT a, b, c, rowtime " +
+      "FROM sourceTable order by rowtime, a desc"
+    tEnv.sqlUpdate(sql)
+    env.execute()
+
+    val expected = List(
+      "1,1,Hi,1970-01-01 00:00:00.001",
+      "3,2,Hello world,1970-01-01 00:00:00.002",
+      "2,2,Hello,1970-01-01 00:00:00.002")
+    assertEquals(expected.sorted, MemoryTableSourceSinkUtil.tableDataStrings.sorted)
```
[the remark this answers is not in the document]
Comment (xueyumusic) [reply flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/sql/SqlITCase.scala]: Thanks! @hequn8128 , updated the code.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/sql/validation/SortValidationTest.scala [hunk 3]
```diff
@@ -54,4 +53,12 @@ class SortValidationTest extends TableTestBase {
     val sqlQuery = "SELECT a FROM MyTable LIMIT 3"
     streamUtil.verifySql(sqlQuery, "")
   }
+
+  // test should fail because time is not order field
+  @Test(expected = classOf[TableException])
+  def testNonTimeSorting(): Unit = {
+
```
Comment (yanghua) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/sql/validation/SortValidationTest.scala]: Remove this empty line looks better.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/sql/SortITCase.scala [hunk 4]
```diff
@@ -105,6 +107,36 @@ class SortITCase extends StreamingWithStateTestBase {
...
+
+  @Test
+  def testInsertIntoMemoryTableOrderBy(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    MemoryTableSourceSinkUtil.clear()
+
+    val t = StreamTestData.getSmall3TupleDataStream(env)
+        .assignAscendingTimestamps(x => x._2)
```
Comment (yanghua) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/sql/SortITCase.scala]: The "." aligns with the next line looks better?

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/sql/SortITCase.scala [hunk 5]
```diff
@@ -105,6 +107,36 @@ class SortITCase extends StreamingWithStateTestBase {
...
+    tEnv.registerTable("sourceTable", t)
+
+    val fieldNames = Array("d", "e", "f", "t")
+    val fieldTypes = Array(Types.INT, Types.LONG, Types.STRING, Types.SQL_TIMESTAMP)
+      .asInstanceOf[Array[TypeInformation[_]]]
+    val sink = new MemoryTableSourceSinkUtil.UnsafeMemoryAppendTableSink
+    tEnv.registerTableSink("targetTable", fieldNames, fieldTypes, sink)
+
+    val sql = "INSERT INTO targetTable SELECT a, b, c, rowtime " +
+      "FROM sourceTable order by rowtime, a desc"
```
Comment (yanghua) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/sql/SortITCase.scala]: SQL keyword (`ORDER BY`) upper case looks better to me.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/sql/validation/SortValidationTest.scala [hunk 6]
```diff
@@ -54,4 +53,12 @@ class SortValidationTest extends TableTestBase {
     val sqlQuery = "SELECT a FROM MyTable LIMIT 3"
     streamUtil.verifySql(sqlQuery, "")
   }
+
+  // test should fail because time is not order field
+  @Test(expected = classOf[TableException])
+  def testNonTimeSorting(): Unit = {
```
Comment (twalthr) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/sql/validation/SortValidationTest.scala]: This test is not testing what it is supposed to test. I will remove it.

**Pull request closed without merging**